api/fluxApplication/FluxApplicationRestHandler.go

```diff
@@ -0,0 +1,88 @@
+package fluxApplication
+
+import (
+	"context"
+	"errors"
+	"github.com/devtron-labs/devtron/api/restHandler/common"
+	"github.com/devtron-labs/devtron/pkg/auth/authorisation/casbin"
+	clientErrors "github.com/devtron-labs/devtron/pkg/errors"
+	"github.com/devtron-labs/devtron/pkg/fluxApplication"
+	"github.com/gorilla/mux"
+	"go.uber.org/zap"
+	"net/http"
+)
+
+type FluxApplicationRestHandler interface {
+	ListFluxApplications(w http.ResponseWriter, r *http.Request)
+	GetApplicationDetail(w http.ResponseWriter, r *http.Request)
+}
+
+type FluxApplicationRestHandlerImpl struct {
+	fluxApplicationService fluxApplication.FluxApplicationService
+	logger                 *zap.SugaredLogger
+	enforcer               casbin.Enforcer
+}
+
+func NewFluxApplicationRestHandlerImpl(fluxApplicationService fluxApplication.FluxApplicationService,
+	logger *zap.SugaredLogger, enforcer casbin.Enforcer) *FluxApplicationRestHandlerImpl {
+	return &FluxApplicationRestHandlerImpl{
+		fluxApplicationService: fluxApplicationService,
+		logger:                 logger,
+		enforcer:               enforcer,
+	}
+
+}
+
+func (handler *FluxApplicationRestHandlerImpl) ListFluxApplications(w http.ResponseWriter, r *http.Request) {
+
+	//handle super-admin RBAC
+	token := r.Header.Get("token")
+	if ok := handler.enforcer.Enforce(token, casbin.ResourceGlobal, casbin.ActionGet, "*"); !ok {
+		common.WriteJsonResp(w, errors.New("unauthorized"), nil, http.StatusForbidden)
+		return
+	}
+	v := r.URL.Query()
+	clusterIdString := v.Get("clusterIds")
+	var clusterIds []int
+	var err error
+
+	//handling when the clusterIds string is empty ,it will not support the
+	if len(clusterIdString) == 0 {
+		handler.logger.Errorw("error in getting cluster ids", "error", err, "clusterIds", clusterIds)
+		common.WriteJsonResp(w, errors.New("error in getting cluster ids"), nil, http.StatusBadRequest)
+	}
+	clusterIds, err = common.ExtractIntArrayQueryParam(w, r, "clusterIds")
+	if err != nil {
+		handler.logger.Errorw("error in getting cluster ids", "error", err, "clusterIds", clusterIds)
+		return
+	}
+	handler.fluxApplicationService.ListFluxApplications(r.Context(), clusterIds, w)
+}
+
+func (handler *FluxApplicationRestHandlerImpl) GetApplicationDetail(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	clusterIdString := vars["appId"]
+	appIdentifier, err := handler.fluxApplicationService.DecodeFluxAppId(clusterIdString)
```

Comment:
We can directly use DecodeFluxExternalAppAppId from helper function, no need to bind the method in interface and expose it, since it's just a helper function.